### PR TITLE
Setup boilerplate for recieving slack commands for bots

### PR DESCRIPTION
**Pull request by joshblum**

### CODE REVIEW

File: example_project/example_project/orchestra_settings.py [hunk 1]
```diff
@@ -190,6 +190,10 @@ def setup_orchestra(settings_module_name):
     settings.SLACK_EXPERTS_API_KEY = ''
     settings.SLACK_INTERNAL_NOTIFICATION_CHANNEL = '#orchestra-tasks'
 
+    # Settings for orchestra bots. Each bot needs a slack token to
```
Comment (kkamalov) [inline example_project/example_project/orchestra_settings.py]: mention in comments what type of bot is it. (i.e. slash command token)

File: example_project/example_project/orchestra_settings.py [hunk 2]
```diff
@@ -190,6 +190,10 @@ def setup_orchestra(settings_module_name):
...
+    # authenticate the command
+    settings.STAFFBOT_TOKEN = ''
```
Comment (kkamalov) [inline example_project/example_project/orchestra_settings.py]: SLACK_STAFFBOT_TOKEN ?

File: orchestra/bots/staffbot.py [hunk 3]
```diff
@@ -1,8 +1,75 @@
...
+        if token != self.token:
+            raise SlackCommandInvalidRequest(
+                'Token mismatch {} != {}'.format(token, self.token))
+
+        for fieldname, whitelist in self.whitelists.items():
+            if whitelist is None:
+                continue
+            else:
+                try:
+                    value = data[fieldname]
```
Comment (kkamalov) [inline orchestra/bots/staffbot.py:47]: `hasattr(data, fieldname)`

File: orchestra/bots/tests/test_staffbot.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+from orchestra.tests.helpers import OrchestraTestCase
+from orchestra.bots.errors import SlackCommandInvalidRequest
+from orchestra.bots.staffbot import StaffBot
+from orchestra.bots.tests.fixtures import get_mock_slack_data
+
+
+@override_settings(STAFFBOT_TOKEN='test-token')
+class StaffBotTests(OrchestraTestCase):
+
+    def test_request_validation(self):
```
Comment (kkamalov) [inline orchestra/bots/tests/test_staffbot.py:11]: Should this be a BotTest not StaffBotTest?

File: example_project/example_project/orchestra_settings.py (hunk 1 above)
Comment (joshblum) [inline example_project/example_project/orchestra_settings.py]: done

File: example_project/example_project/orchestra_settings.py (hunk 2 above)
Comment (joshblum) [inline example_project/example_project/orchestra_settings.py]: done

File: orchestra/bots/staffbot.py (hunk 3 above)
Comment (joshblum) [inline orchestra/bots/staffbot.py:47]: done

File: orchestra/bots/tests/test_staffbot.py (hunk 4 above)
Comment (joshblum) [inline orchestra/bots/tests/test_staffbot.py:11]: done

**Pull request merged**